### DISCUSSION TITLE
Plan for OpenTelemetry

### DISCUSSION THREAD

**Discussion by alberk8** [nats-io/nats.net#624] [Q&A]
May I ask what are the plan for OpenTelemetry?  I do see a [branch here](https://github.com/nats-io/nats.net/tree/stage-otel).

**Comment by mtmk** [comment]
we support tracing and metrics support is in the backlog. that's an old version of a branch merged while back.

**Reply by alberk8** [reply]
Thanks. Would be nice to an AutoInstrumentation like `AddNatsInstrumenation`

**Reply by mtmk** [reply]
thanks for the suggestion. do you have an example library doing similar?

**Reply by alberk8** [reply]
This is [AspnetCore and Http example](https://github.com/open-telemetry/opentelemetry-dotnet/blob/685b73d3ac7a4d4f0b6d658650f93fb1ad166e40/examples/AspNetCore/Program.cs#L51)
[Source Code](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Instrumentation.AspNetCore)
